evap/staff/views.py

```diff
@@ -144,159 +149,194 @@ def __init__(self):
+
+    @staticmethod
+    def applicable_to(evaluation):
+        return evaluation.state in ['prepared', 'editor_approved', 'approved']
+
+    @staticmethod
+    def warning_for_inapplicables(amount):
+        return ungettext("{} evaluation can not be reverted, because it already started. It was removed from the selection.",
+            "{} evaluations can not be reverted, because they already started. They were removed from the selection.", amount).format(amount)
+
+    @staticmethod
+    def apply(request, evaluations, email_template=None):
+        for evaluation in evaluations:
+            evaluation.revert_to_new()
+            evaluation.save()
+        messages.success(request, ungettext("Successfully reverted {} evaluation to in preparation.",
+            "Successfully reverted {} evaluations to in preparation.", len(evaluations)).format(len(evaluations)))
+
+
+class RevertToPreparedOperation(EvaluationOperation):
+    email_template_name = EmailTemplate.EDITOR_REVIEW_NOTICE
+    confirmation_message = ugettext_lazy("Do you want to revert the following evaluations to preparation?")
+
+    @staticmethod
+    def applicable_to(evaluation):
+        return evaluation.state in ['new', 'editor_approved']
+
+    @staticmethod
+    def warning_for_inapplicables(amount):
+        return ungettext("{} evaluation can not be reverted, because it already started. It was removed from the selection.",
+            "{} evaluations can not be reverted, because they already started. They were removed from the selection.", amount).format(amount)
+
+    @staticmethod
+    def apply(request, evaluations, email_template=None):
+        for evaluation in evaluations:
+            evaluation.ready_for_editors()
+            evaluation.save()
+        messages.success(request, ungettext("Successfully enabled {} evaluation for editor review.",
+            "Successfully enabled {} evaluations for editor review.", len(evaluations)).format(len(evaluations)))
+        if email_template:
+            EmailTemplate.send_to_users_in_evaluations(email_template, evaluations, [EmailTemplate.EDITORS], use_cc=True, request=request)
+
+
+class StartEvaluationOperation(EvaluationOperation):
+    email_template_name = EmailTemplate.EVALUATION_STARTED
+    confirmation_message = ugettext_lazy("Do you want to immediately start the following evaluations?")
+
+    @staticmethod
+    def applicable_to(evaluation):
+        return evaluation.state == 'approved' and evaluation.vote_end_date >= date.today()
+
+    @staticmethod
+    def warning_for_inapplicables(amount):
+        return ungettext("{} evaluation can not be started, because it was not approved, was already evaluated or its evaluation end date lies in the past. It was removed from the selection.",
+            "{} evaluations can not be started, because they were not approved, were already evaluated or their evaluation end dates lie in the past. They were removed from the selection.", amount).format(amount)
```

Comment:
i'd suggest to make `warning_for_inapplicables` a property just like `confirmation_message` is.

Reply:
I wasn't sure how to do this, as this is using the pluralized form. How do I make a property out of that?

Reply:
um, yeah, i didn't realize this was two strings in a method call. i can't think of something that doesn't look ugly.